go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -490,7 +490,14 @@ func (tsv *TabletServer) begin(ctx context.Context, target *querypb.Target, save
 		target, options, false, /* allowOnShutdown */
 		func(ctx context.Context, logStats *tabletenv.LogStats) error {
 			startTime := time.Now()
-			if tsv.txThrottler.Throttle() {
+			criticality := tsv.config.TxThrottlerDefaultCriticality
+			if options != nil && options.Criticality != "" {
+				optionsCriticality, err := strconv.Atoi(options.Criticality)
+				if err == nil {
+					criticality = optionsCriticality
+				}
```

Comment:
I am unsure I follow. If the value is not a valid integer, it will give an error, so criticality will preserve the default value set above instead. Or am I misunderstanding what you mean?

Reply:
What I mean is if the value is not an integer, we should return an error, we should break the flow. As it is right now, an invalid value is not even logged. What I'd like to see is `if err != nil { return "", err}`.

Reply:
Since we added all the validations on the `vtgate` side (both that it can be parsed to integer and that it is within range, and otherwise it fails the query) I am tempted to remove error checking altogether here. Thoughts?

Reply:
Looks like errors are now logged.

Reply:
Fixed